pkg/stub/config.go

```diff
@@ -170,34 +170,43 @@ func (h *Handler) VariableConfigChanged(cfg *v1.Config) (bool, bool, bool) {
 }
 
 func (h *Handler) buildSkipFilters(opcfg *v1.Config) {
-	newStreamMap := make(map[string]bool)
-	newTempMap := make(map[string]bool)
-	for _, st := range opcfg.Spec.SkippedTemplates {
-		newTempMap[st] = true
-	}
-	for _, si := range opcfg.Spec.SkippedImagestreams {
-		newStreamMap[si] = true
+	h.mapsMutex.Lock()
+	defer h.mapsMutex.Unlock()
+	if len(opcfg.Spec.SkippedImagestreams) != len(h.skippedImagestreams) ||
+		len(opcfg.Spec.SkippedTemplates) != len(h.skippedTemplates) {
```

Comment:
don't you also have to care if the values changed within the arrays (ie one value was replaced by another, to the lengths stayed the same?)

Reply:
yeah with the mutex and rechecking the remaining calls, removing these checks and rebuilding is fine

will just simplify that way